### DISCUSSION TITLE
How does targets decide when to invalidate a targeting involving an online API call

### DISCUSSION THREAD

**Discussion by liutiming** [ropensci/targets#440] [Help] [answered]
I have a function that calls an online API that involves computation from the value that I provided. 

I wonder how hash value is determined in this case and if it is recommended to set `tar_cue = "never"` once the call has been done.

**Comment by wlandau** [comment, marked as answer]
The hash is determined by the local data that is downloaded. The target is invalidated when the local data changes, etc., but `targets` will not know about the external service whose API you called. Up to you whether to set `mode = "never"` in `tar_cue()`. If all you need is the data to exist and you do not care to synchronize your local copy with the latest online copy, that seems fine. Otherwise, you could use a custom cue, maybe a `tar_age()` target or `tar_cue_age()` cue from development `tarchetypes` if you want to check at regular time intervals.

**Reply by liutiming** [reply]
Thanks a lot!
